### ISSUE TITLE
PR creation doesn't recognize remote branch

### ISSUE THREAD

**Issue by dominicmeyer** [pwntester/octo.nvim#400]
## Issue Description

*bug report*

### Describe what happened (or what feature you want)

I wanted to create a pull request with octo, but for some reason there was a message displayed, that the current branch has no remote and if I wanted to push the current one. The current brach was pulled from gh and also did pushes right before, so it was clearly connected. After restarting nvim I was able to create the pr.

### Describe what you expected to happen

The pr creation was successful most of the time, so I didn't expect it to fail. Of course restarting nvim isn't that big of a deal, but it's just not convenient.

### How to reproduce it (as minimally and precisely as possible)

I had the bug before, but sadly I don't know how to recreate it yet.

### Tell us your environment

Windows 11
gh version 2.29.0
octo latest

### Anything else we need to know?

Maybe this is an issue of gh cli?

**Comment by pwntester** [comment]
Thanks for reporting the issue and sorry for the late response. Are you still experiencing this bug? Remote branches are retrieved from the GH API. It is possible that if this request fails for whatever reason, it defaults to think that the remote does not exist. I just pushed a change to errout if the request fails. Please reopen if the problem persists

**Comment by Dronakurl** [comment]
I also ran into this issue. How can I check if the gh API returns my remote branch?

**Comment by wd60622** [comment]
> I also ran into this issue. How can I check if the gh API returns my remote branch?

Can you try with the REST API ? https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#merge-a-branch

**Comment by Dronakurl** [comment]
yes this did work:

```bash
curl -s https://api.github.com/repos/Dronakurl/muxbar/branches
```

```json
[
  {
    "name": "backup",
    "commit": {
      "sha": "40be0d5e46d4ad3bd0cf4160ba28be85e1fab417",
      "url": "https://api.github.com/repos/Dronakurl/muxbar/commits/40be0d5e46d4ad3bd0cf4160ba28be85e1fab417"
    },
    "protected": false
  },..
  ```

**Comment by wd60622** [comment]
> yes this did work:
> 
> 
> 
> ```bash
> 
> curl -s https://api.github.com/repos/Dronakurl/muxbar/branches
> 
> ```
> 
> 
> 
> ```json
> 
> [
> 
>   {
> 
>     "name": "backup",
> 
>     "commit": {
> 
>       "sha": "40be0d5e46d4ad3bd0cf4160ba28be85e1fab417",
> 
>       "url": "https://api.github.com/repos/Dronakurl/muxbar/commits/40be0d5e46d4ad3bd0cf4160ba28be85e1fab417"
> 
>     },
> 
>     "protected": false
> 
>   },..
> 
>   ```

Would you like to make a PR?

**Comment by Dronakurl** [comment]
Sorry, I mean, I didn't get it to work in Octo.nvim, but the REST API of github returns that there is a remote branch of the same name.

**Comment by wd60622** [comment]
> Sorry, I mean, I didn't get it to work in Octo.nvim, but the REST API of github returns that there is a remote branch of the same name. 

I know. Would you like to make a PR to fix the bug?

**Comment by Dronakurl** [comment]
Yes, would be happy to contribute. It might take some time, though

**Comment by wd60622** [comment]
> Yes, would be happy to contribute. It might take some time, though

Great! Thanks 
No rush. Open a draft PR if needed

**Comment by Dronakurl** [comment]
Can you please review my draft?
